src/components/FilterBar.tsx

```diff
@@ -72,6 +67,25 @@ const FilterBar: React.FC<IFilterBarProps> = (props) => {
     [selectedFilters, updateSelectedFilters]
   );
 
+  const filterBarOptions = [];
+  const filterPropertyNames = Object.keys(filterOptions);
+  
+  //console.log("filterOptions ", filterOptions);
+
+  for (var i = 0; i < filterPropertyNames.length; i++) {
+    const filterPropertyName = filterPropertyNames[i];
+    const filterValues = filterOptions[filterPropertyName];
+    const name = ProjectFilterProperties[filterPropertyName];
+    const label = "Filter by " + name;
+    filterBarOptions.push(
+      <FilterBarOptions
+      filters={filterValues}
+      label={label} 
+      allowCollapse={allowCollapse}
+      addAction={handleFilterAddClick}
+    />
```

Comment:
NIT: This is a fine way to do this, but it isn't what I would refer to as "idiomatic" React. Generally you want to keep all of your component declarations in the `return` JSX block. See my following comment...